deep-learning/src/test/scala/com/microsoft/azure/synapse/ml/cntk/ImageFeaturizerSuite.scala

```diff
@@ -151,36 +121,37 @@ class ImageFeaturizerSuite extends TransformerFuzzing[ImageFeaturizer]
     val testImg: DataFrame = spark
       .read.image.load(s"$filesRoot/Images/Grocery/testImages/WIN_20160803_11_28_42_Pro.jpg")
       .withColumnRenamed("image", inputCol)
-    val result = resNetModel().transform(testImg)
+    val result = resNetModelFull().transform(testImg)
     val resVec = result.select(outputCol).collect()(0).getAs[DenseVector](0)
-    assert(resVec.argmax == 760)
+    val top2 = argtopk(resVec.toBreeze, 2).toArray
+    assert(top2.contains(760))
   }
 
   test("Image featurizer should work with ResNet50 and powerBI") {
     val images = groceryImages.withColumnRenamed(inputCol, "image").coalesce(1)
     println(images.count())
 
-    val result = resNetModel().setInputCol("image").transform(images)
+    val result = resNetModelFull().setInputCol("image").transform(images)
       .withColumn("foo", UDFUtils.oldUdf({ x: DenseVector => x(0).toString }, StringType)(col("out")))
       .select("foo")
 
-    PowerBIWriter.write(result,sys.env.getOrElse("MML_POWERBI_URL", Secrets.PowerbiURL), Map("concurrency" -> "1"))
-  }
-
-  test("test layers of network") {
-    (0 to 9).foreach({ i =>
-      val model = new ImageFeaturizer()
-        .setModel(resNet)
-        .setInputCol(inputCol)
-        .setOutputCol(outputCol)
-        .setCutOutputLayers(i)
-      val result = model.transform(images)
-    })
+    PowerBIWriter.write(result, sys.env.getOrElse("MML_POWERBI_URL", Secrets.PowerbiURL), Map("concurrency" -> "1"))
   }
 
   val reader: MLReadable[_] = ImageFeaturizer
 
   override def testObjects(): Seq[TestObject[ImageFeaturizer]] = Seq(
-    new TestObject(resNetModel(), images)
-  )
+      new TestObject(resNetModelHeadless(), images)
+    )
+
+  // Override python objects because modelPayload of ONNXModel is a ComplexParam, which
+  // cannot be used in python.  Use simple ImageFeaturizer.
+  override def pyTestObjects(): Seq[TestObject[ImageFeaturizer]] = {
+    val testObject: ImageFeaturizer = new ImageFeaturizer()
+      .setInputCol(inputCol)
```

Comment:
Does this default to using the model string API? or is the model not set here?

Reply:
if we set the model we get an error due to the ComplexParam.  Is there a known solution for this?

Reply:
note that setting the model as a string (the onnxhub name) cause the model bytes to be set under the covers

Reply:
You can implement the APIs for pythron wrapping that complex param in the tests. Shoulkd be something like extending PyParamWrappable or something like that. Dont worry if you want to leave this for later